### PR TITLE
Send calendar events by mail

### LINKED ISSUE #450: Send calendar events by mail
Use #417

### PR DESCRIPTION
**Pull request by nyet-ty**
Resolve #450

### CODE REVIEW

File: src/modules/sectionMethods.ts [hunk 1]
```diff
@@ -80,7 +87,39 @@ const create = async (data: CreateSection): Promise<Section> => {
...
+        const interviewer = await prisma.user.findFirstOrThrow({ where: { id: interviewerId } });
+
+        const icalEventDataException = createIcalEventData({
+            id: newSection.calendarSlotId,
+            users: [{ email: interviewer.email, name: interviewer.name || undefined }],
+            start: exception.date,
+            description: restData.description || exception.eventDetails.description,
+            duration: exception.eventDetails.duration,
+            summary: `Interview with ${interview.candidate.name}`,
+            url: `${config.defaultPageURL}/interviews/${interviewId}/sections/${newSection.id}`,
```
Comment (9teen90nine) [inline src/modules/sectionMethods.ts:107]: Don't we have a helper for routes? @tsumo

File: src/trpc/routers/calendarEventsRouter.ts [hunk 2]
```diff
@@ -37,20 +37,23 @@ export const calendarEventsRouter = router({
...
             switch (part) {
                 case 'exception':
-                    return calendarEventMethods.createEventException(input);
+                    return calendarEventMethods.createEventException(input, ctx.session.user);
                 case 'future':
-                    return calendarEventMethods.splitEventSeries(input, ctx.session.user.id);
+                    return calendarEventMethods.splitEventSeries(input, ctx.session.user);
                 case 'series':
-                    return calendarEventMethods.updateEventSeries(input);
+                    return calendarEventMethods.updateEventSeries(input, ctx.session.user);
```
Comment (9teen90nine) [inline src/trpc/routers/calendarEventsRouter.ts:56]: you probably want calendar event owner here, not the user?

Comment (nyet-ty) [reply src/trpc/routers/calendarEventsRouter.ts:56]: yes, you are right, it may be not the same user

File: src/modules/sectionMethods.ts (hunk 1 above)
Comment (tsumo) [reply src/modules/sectionMethods.ts:107]: we do, `utils/paths`

Comment (nyet-ty) [reply src/modules/sectionMethods.ts:107]: I'll edit it here and in notifyHr

File: src/trpc/routers/calendarEventsRouter.ts (hunk 2 above)
Comment (nyet-ty) [reply src/trpc/routers/calendarEventsRouter.ts:56]: event creator can be null, so user here also needed

**Review by 9teen90nine — APPROVED** [review]

**Review by tsumo — APPROVED** [review]

**Pull request merged**